src/utils/middleware.ts

```diff
@@ -18,6 +19,31 @@ import { IDPEntityDescriptor } from "../types/IDPEntityDescriptor";
 import { fetchIdpsMetadata } from "./metadata";
 import { logSamlCertExpiration, SamlAttributeT } from "./saml";
 
+export enum ProfessionalPurpose {
+  P = "P",
+  PX = "PX",
+  LP = "LP",
+  PG = "PG",
+  PF = "PF"
+}
+
+const ProfessionalSpidExtension = t.union([
+  t.interface({
+    professionalSpidEnabled: t.literal(false)
+  }),
+  t.interface({
+    professionalSpidEnabled: t.literal(true),
+    purpose: t.union([
+      t.literal(ProfessionalPurpose.P),
+      t.literal(ProfessionalPurpose.PX),
+      t.literal(ProfessionalPurpose.LP),
+      t.literal(ProfessionalPurpose.PG),
+      t.literal(ProfessionalPurpose.PF)
+    ])
+  })
+]);
```

Comment:
If you need to consider every occurrence of `ProfessionalPurpose` in `ProfessionalSpidExtension` codec, you can link them by trasforming the first into a string literal union:
```suggestion
type ProfessionalPurpose = t.TypeOf<typeof ProfessionalPurpose>;
export const ProfessionalPurpose = t.keyof({
  P: null,
  PX: null,
  LP: null,
  PG: null,
  PF: null,
});

const ProfessionalSpidExtension = t.union([
  t.interface({
    professionalSpidEnabled: t.literal(false)
  }),
  t.interface({
    professionalSpidEnabled: t.literal(true),
    purpose: ProfessionalPurpose
  })
]);
```